controllers/runner_controller.go

```diff
@@ -112,6 +114,34 @@ func (r *RunnerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 			// Pod was not found
 			return r.processRunnerDeletion(runner, ctx, log, nil)
 		}
+		// remove runner-linked pods
+		var runnerLinkedPodList corev1.PodList
+		r.List(ctx, &runnerLinkedPodList, client.MatchingLabels(
```

Comment:
do we need to check the state? aka only try to delete pods that aren't already deleting?

Reply:
Do we also need a retry behavior for this, like we have for getting the runner pod above?

Reply:
Thanks, I will add that as well!